Center the selected item when a select-shaped popup opens (#7067)
ariakit/ariakit: .changeset/300-center-selected-popup-item.md

---
"@ariakit/react": patch
"@ariakit/react-components": patch
---

> Centered the initially selected item when a [`ComboboxSelect`](https://ariakit.com/reference/combobox-select) popup opened while keeping later keyboard, typeahead, and pointer navigation nearest-edge aligned.

Author: <!-- ariakit-pr-review review_id=pr-7067-6817903bd8ecee481f867a422842fd0971a511b4-20260804T172407Z head_sha=6817903bd8ecee481f867a422842fd0971a511b4 -->

Two things about this entry.

It reads as the opposite of `200-dialog-initial-focus-scroll.md`, which is unreleased in the same batch: "Changed `Dialog` to scroll its initially focused element into view by the shortest distance that makes it visible, rather than centering it, which is visible in `ComboboxSelect` and `SelectItem` where a selected value below the fold now rests against the edge of its list rather than near the middle." A reader of the next changelog gets both bullets and no way to tell which one describes what they will see. Worth reconciling the wording so the pair reads as one story: `Dialog` no longer centers, and select-shaped popups center their selected value themselves on open.

It also understates the scope. The centering branch keys off `state.selectElement`, which the deprecated `Select` store also carries, so `Select` plus `SelectPopover` centers too, through `SelectItem`'s `data-autofocus` marker and the same marked presentation. That is a reasonable thing to do, but it is user-visible behavior the entry does not mention and the tests do not cover.

**Assessment**

- Severity: Low (verified, 90% confidence), because it is release-notes accuracy rather than runtime behavior.
- Priority: P2 (estimate, 80% confidence), because it should be right before the release goes out but does not block the code.
- Breakage likelihood: Likely (verified, 90% confidence), because both entries are already queued for the same release.
- Fix effort: Easy (verified, 95% confidence), because it is a wording change.

Author: Addressed. The Dialog and ComboboxSelect entries now tell one consistent release story, and each changeset description is a single user-facing sentence without implementation details.

Author: Addressed. The Dialog and ComboboxSelect entries now tell one consistent release story, and each changeset description is a single user-facing sentence without implementation details.

> Changed [`ComboboxSelect`](https://ariakit.com/reference/combobox-select) to center the initially selected item when its popup opens, including default-open popups and asynchronously rendered options, while preserving nearest-edge scrolling during later navigation.

Author: Changed -> Fixed

Also, it doesn't need too much details: the default-open, etc. are all unnecessary.

Author: <!-- ariakit-pr-review review_id=pr-7067-4431961df860fd52c18f77817776077bbd72276e-20260805T095731Z head_sha=4431961df860fd52c18f77817776077bbd72276e -->
Confirmed and still unaddressed at `4431961d`. The entry currently reads:

> Changed [`ComboboxSelect`](https://ariakit.com/reference/combobox-select) to center the initially selected item when its popup opens, including default-open popups and asynchronously rendered options, while preserving nearest-edge scrolling during later navigation.

Both points still apply: it opens with `Changed`, and the trailing clauses describe the implementation's coverage rather than what a reader needs in order to know whether the change affects them.

```suggestion
Fixed [`ComboboxSelect`](https://ariakit.com/reference/combobox-select) to center the initially selected item when its popup opens.
```

That also matches the repository's simple-entry shape (`Fixed <API> to <behavior>.`) and stays a single past-tense sentence. No contributor credit sentence is needed here: issue #7011 was opened by the maintainer and its only other participants are bots.

**Assessment**

- Severity: Low (verified, evidence: changelog copy only, no runtime effect).
- Priority: P1 (estimate, 90% confidence), because it is a one-line copy change already requested on this thread.
- Breakage likelihood: Likely (verified, evidence: the text ships to the changelog as written).
- Fix effort: Easy (verified, evidence: single line, no code or tests involved).

Author: Addressed in [`af641d55c`](https://github.com/ariakit/ariakit/commit/af641d55c). The entry now starts with `Fixed` and says only that `ComboboxSelect` centers the initially selected item when its popup is opened interactively.
